### DISCUSSION TITLE
API Data is being passed but not being rendered

### DISCUSSION THREAD

**Discussion by arimtiaz** [vercel/next.js#65143] [Help]
### Summary

I am using nextjs and woocommerce rest api to build a ecommerce site. But I have been facing issues with rendering products/data from woocommerce to nextjs frontend. I have setup the endpoint for getting the products and it is being shown in the console that they are passed, but whenever I try to render them on the home page using map it says no product.

Console: https://prnt.sc/DqBK_QZv3Lfv

`{
import axios from "axios";
import { Albert_Sans, Manrope } from "next/font/google";
import Navbar from "./Blocks/Navbar";
import Hero from "./Component/Hero";
import InfiniteCarousel from "./Component/InfiniteCarousel";
import Logos from "./Component/Logos";
import Category from "./Component/Category";
import("next").InferGetStaticPropsType;

const albert = Albert_Sans({ subsets: ["latin"], weight: ["400", "700"] });
const manrope = Manrope({ subsets: ["latin"] });

export async function getStaticProps() {
  try {
    const { data: products } = await axios.get(
      "http://localhost:3000/api/get-products"
    );

    const data = { products: products || [] };

    return {
      props: data,
    };
  } catch (error) {
    console.error("Error fetching products:", error);
    return {
      props: {
        products: [],
      },
    };
  }
}

export default function Home({ products }) {
  console.log("Products received in Home component:", products);

  return (
    <main
      className={`flex min-h-screen mx-auto bg-dark-shade-1 flex-col items-center ${manrope.className}`}
    >
      <Navbar />
      <Hero />
      <InfiniteCarousel className="w-full" />
      <Logos />
      <Category />
      {products.length > 0 ? (
        <div className="products-container">
          {products.map((product) => (
            <div key={product.id} className="product">
              <h3 className="text-white text-h2">{product.name}</h3>
              <p className="text-white text-l2">{product.price}</p>
            </div>
          ))}
        </div>
      ) : (
        <p className="text-white">No Products Found</p>
      )}
    </main>
  );
}

}`

**Comment by icyJoseph** [comment]
### Perhaps the problem

I think you are just not dealign with the data shape you think you are.

From your screenshot, products is:

```json
{
   "product": [ /* list of products */],
   "success":
}
```

So you have to do:

```js
products.product.map((entry) => (/* do something with this entry */))
```

### Maybe not the problem, but a very big problem

There's already a very big problem, and is that you are, sort of, using getStaticProps incorrectly. It is not showing now, but when you try to build the app, it will crash. 

Your GetStaticProps is fetching from API routes, which are not available at build time. You can read this answer from the Discord FAQ, to get more sense into the issue: https://nextjs-faq.com/fetch-api-in-getserversideprops ~ though it talks about (GetServerSideProps) GSSP, it also applies to GetStaticProps(GSP), just make the call to your API service directly, no need to do an extra jump. 

And if you are worried about leaking code to the client, take a look at:
  
- https://next-code-elimination.vercel.app/
